### DISCUSSION TITLE
v1.2.0 Released

### DISCUSSION THREAD

**Discussion by Makuna** [Makuna/DFMiniMp3#117] [Announcements]
The delay mechanism has been replaced with a more robust communications transaction model.  Most of this happens behind the scenes so there is little change from the sketch writers other than more stable operations.

* Coms now check for Ack/Error and will retry several times before they notify an error has happened.  
* Notifications are queued and called so they do not interfere with the communications transactions.  
* `getSoftwareVersion()` has been added that will query the module for its software version.  
* For platforms that allow for the hardware serial device to work on arbitrary pins (ESP32 as example) a constructor has been added that takes the rxPin and txPin.

**Comment by FranzKropp** [comment]
I found the getStatus() function has changed! Just upgrading the library leed to strange Error Messages. It would be helpfull to place this function in one of the examples.

Version 1.x: dfmp3.getStatus() 

Version 2.x: dfmp3.getStatus().state  or dfmp3.getStatus().source is needed.

**Reply by Makuna** [reply]
https://github.com/Makuna/DFMiniMp3/issues/121 to track this.
Yes, previously the method was marked as unknown return values and not documented in specs well.  It has some chip specific nuances to it mostly around sleep state. I found updated documents and tested them and updated the method.  

[getStatus() Wiki entry](https://github.com/Makuna/DFMiniMp3/wiki/API-Reference#dfmp3_status-getstatus)
